parser/declaration.go

```diff
@@ -1636,13 +1594,7 @@ func parseMemberOrNestedDeclaration(p *parser, docString string) (ast.Declaratio
 				)
 
 			case KeywordCase:
-				if purity != ast.FunctionPurityUnspecified {
-					return nil, NewSyntaxError(*purityPos, "invalid view modifier for enum case")
-				}
-				err := rejectStaticAndNativeModifiers(p, staticPos, nativePos, common.DeclarationKindEnumCase)
-				if err != nil {
-					return nil, err
-				}
+				rejectNonAccessModifiers(p, staticPos, nativePos, purityPos, common.DeclarationKindEnumCase)
```

Comment:
Is it intentional that now the parser would continue to parse, in case of an error?